doc/api/typing_api.rst

```diff
@@ -2,18 +2,33 @@
 ``matplotlib.typing``
 *********************
 
+.. automodule:: matplotlib.typing
+   :no-members:
+   :no-undoc-members:
+
+Color
+=====
+
+.. autodata:: matplotlib.typing.ColorType
 .. autodata:: matplotlib.typing.RGBColorType
-.. autodata:: matplotlib.typing.RGBColourType
 .. autodata:: matplotlib.typing.RGBAColorType
-.. autodata:: matplotlib.typing.RGBAColourType
-.. autodata:: matplotlib.typing.ColorType
 .. autodata:: matplotlib.typing.ColourType
+.. autodata:: matplotlib.typing.RGBColourType
+.. autodata:: matplotlib.typing.RGBAColourType
```

Comment:
These seem to need docstrings, even if they are just "Alias of ...", because right now they are the rather unclear "Represent a PEP 604 union type".

Also, any idea why these colour types seem to get the type alias in bold next to the name, whereas the style ones seem to get "alias of ..." in the description instead?

Reply:
I've noted this as well, but decided to not bother right now. I don't want to spend time on it, and it's not getting worse through the PR.

There must be some issue in autodoc. If that gets fixed, we should get the right description automatically.

More fundamentally, why do we need these aliases? All our function/parameter names use "color" and we don't alias them.

Reply:
If I had my way, I'd have the colour spelling available everywhere, but I don't remember why these specifically are here.

Reply:
I've add docstrings to the aliases for now. Though, I'd propose to remove the aliases because AFAIK we don't have aliases anywhere else in the API.